packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -718,10 +694,17 @@ export class LocalFluidDataStoreContextBase extends FluidDataStoreContext {
     }
 
     public generateAttachMessage(): IAttachMessage {
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        const entries = this.channel!.getAttachSnapshot();
+        assert(this.channel !== undefined, "There should be a channel when generating attach message");
 
-        const snapshot: ITree = { entries, id: null };
+        let snapshot: ITree;
+        if (this.channel.getAttachSummary !== undefined) {
+            const summaryTree = this.channel.getAttachSummary();
+            snapshot = convertSummaryTreeToITree(summaryTree.summary);
```

Comment:
Maybe a comment indicating why we need this forking logic here?
Because attach messages are ITrees